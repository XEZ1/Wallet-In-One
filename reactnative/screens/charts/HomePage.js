import React, { useEffect, useState } from "react";
import { useIsFocused } from "@react-navigation/native";
import { StyleSheet, Text, ScrollView, Dimensions, TouchableOpacity, View, } from "react-native";
import { VictoryPie, VictoryBar, VictoryLabel, VictoryContainer, VictoryStack } from "victory-native";

import { useTheme } from "reactnative/src/theme/ThemeProvider";
import { auth_get, auth_post } from "../../authentication";
import { styles } from "reactnative/screens/All_Styles.style.js";

import NoWallets from "./chartComponents/noWallets";
import PieChart from "./chartComponents/pieChart";
import BarChart from "./chartComponents/barChart"
import StackedChart from "./chartComponents/stackedBarChart";
import fixture from "../charts/chartData.json";
import useCryptoWallet from "../crypto_wallet/useCryptoWallet";

export default function HomePage({ navigation }) {
  const {dark, colors, setScheme } = useTheme();
  const isFocused = useIsFocused();
  const [chartType, setChartType] = useState("pie"); // Default chart is pie chart

  const [baseData, setBaseData] = useState(fixture);
  const [data, setNewData] = useState(baseData.all);
  const [pressed, setPressed] = useState(null);
  const { wallets, fetchWallets, removeWallet } = useCryptoWallet();

  // Uncomment to show bank data from backend

  useEffect(() => {
    const fetchData = async () => {
      const response = await auth_get("/graph_data/");
      console.log("fetch graph data", response.status);
      if (response.status == 200) {
        setBaseData(response.body);
        setNewData(response.body.all);
        setPressed(null);
      }
    };
    if (isFocused) {
      fetchData();
    }
    fetchWallets();
  }, [isFocused]);

  const handlePressIn = async (event, datapoint) => {
    var index = datapoint.index;
    
    if (pressed) {
      if (pressed == "Banks"){
        var bankData = baseData["Banks"][index]
        if (bankData.id){
          navigation.navigate('Bank Transactions', {accountID: bankData.id})
          return
        }
      } else if (pressed === "Cryptocurrency from wallets") {
        var cryptoData = baseData["Cryptocurrency"][index]
        var wallet = wallets.find(x => x.id === cryptoData.id)
        if (cryptoData.id) {
          navigation.navigate("Wallet Detail", { item: wallet, value: cryptoData.y, removeWallet: removeWallet })
          return
        }
      }
      else if (pressed == "Stock Accounts") {
        var stockData = baseData["Stock Accounts"][index]
        if (stockData.id) {
          var response = await auth_get(`/stocks/get_account/${stockData.id}/`)
          const res = await auth_get(`/stocks/list_transactions/${stockData.id}/`)
          navigation.navigate("Stock Account Transactions", {
            accountID: stockData.id, 
            accessToken: response.body.access_token, 
            transactions: res.body,
            logo: response.body.logo,
            balance: response.body.balance
          })
        }
        console.log(stockData.id)

      }
      setNewData(baseData.all);
      setPressed(null)
    } else {
      const dataPoint = data[index];
      const name = dataPoint.x
      if (baseData[name]) {
        setNewData(baseData[name]);
      } else {
        // If data for single type of asset does not exist, just display that one asset from all.
        setNewData(baseData.all.filter((val) => val.x.match(name)));
      }
      setPressed(name);
    }
    
  };

  let value = 0;
  data.forEach((jsonObj) => {
    value += jsonObj.y;
  });
  value = value.toFixed(2);

  const list = data.map((val) => val.x);
  const colours = ["pink", "turquoise", "lime", "#FA991C"];

  let spacing = list.length * 60;

  const handleChartTypeChange = (type) => {
    setChartType(type);
  };

  if (value == 0) {
    return (<NoWallets/>);
  } else {
    return (
      <ScrollView
        contentContainerStyle={{
          flexGrow: 1,
          justifyContent: "center",
          alignItems: "center",
          paddingBottom: 20,
          backgroundColor: colors.background,
        }}
        style={styles.container}
      >

        {/* Switch Graph Buttons */}
        <View style={{ flexDirection: "row", justifyContent: "space-around", width: "90%", backgroundColor: "antiquewhite", margin: 20, borderRadius: 30 }}>
          <TouchableOpacity
            style={[
              styles(dark, colors).btn,
              chartType === "pie" && { backgroundColor: "aliceblue" },
            ]}
            onPress={() => handleChartTypeChange("pie")}
          >
            <Text style={styles.chartTypeText}>Pie Chart</Text>
          </TouchableOpacity>
          <TouchableOpacity
            style={[
              styles(dark, colors).btn,
              chartType === "stacked" && { backgroundColor: "aliceblue" },
            ]}
            onPress={() => handleChartTypeChange("stacked")}
          >
            <Text style={styles.chartTypeText}>Stacked Bar Chart</Text>
          </TouchableOpacity>
        </View>

        <Text >{pressed}</Text>

        {chartType == "pie" ? 
          <>
            <PieChart colours={colours} data={data} handlePressIn={handlePressIn}/>
            {BarChart(colours, list, data, colors, spacing, handlePressIn)}
          </>
          : 
            <StackedChart data={baseData} />
        }

        
        
        {pressed ? (
          <TouchableOpacity
            onPress={() => {
              setNewData(baseData.all);
              setPressed(false);
            }}
          >
            <Text style={[styles(dark, colors).button, { color: colors.text }]}>Back</Text>
          </TouchableOpacity>
        ) : (
          ""
        )}
      </ScrollView>
    );
  }
}

