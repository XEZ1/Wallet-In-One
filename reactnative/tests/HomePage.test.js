// HomePage.test.js
import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import HomePage from '../screens/charts/HomePage';
import renderer from 'react-test-renderer';

describe('<HomePage />', () => {
    it('snapshot test', () => {
        const snapshot = renderer.create(<HomePage />).toJSON();
        expect(snapshot).toMatchSnapshot();
      });
  });