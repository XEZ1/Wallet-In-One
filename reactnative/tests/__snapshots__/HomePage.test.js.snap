// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> snapshot test 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flexGrow": 1,
      "justifyContent": "center",
      "paddingBottom": 20,
    }
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View
      className="VictoryContainer"
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "paddingBottom": 10,
          "position": "relative",
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="300"
        bbWidth="750"
        focusable={false}
        height={300}
        meetOrSlice={0}
        minX={0}
        minY={0}
        role="img"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
            Object {
              "flex": 0,
              "height": 300,
              "width": 750,
            },
          ]
        }
        vbHeight={300}
        vbWidth={750}
        width={750}
      >
        <RNSVGGroup>
          <RNSVGRect
            fill={null}
            height="300"
            propList={
              Array [
                "fill",
              ]
            }
            width="750"
            x="0"
            y="0"
          />
          <RNSVGGroup>
            <RNSVGPath
              d="M2.349691592702662,-239.68205005290886A10,10,0,0,1,12.86426207573194,-249.66880213844672A250,250,0,1,1,-103.38432595796823,227.62179409322073A10,10,0,0,1,-108.12896853696343,213.91851797186308L-48.17612308099603,98.14309620666485A10,10,0,0,1,-35.72373405789277,93.40136414935789A100,100,0,1,0,11.226992357002432,-99.36777466873156A10,10,0,0,1,2.3496915927026745,-109.30455213560472Z"
              fill={
                Object {
                  "payload": 4294951115,
                  "type": 0,
                }
              }
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  375,
                  150,
                ]
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                Array [
                  "fill",
                ]
              }
              responsible={true}
            />
            <RNSVGPath
              d="M-112.30202814477764,211.75755156006954A10,10,0,0,1,-126.23129558551211,215.79077833587576A250,250,0,0,1,-243.61354339017154,56.14660699352197A10,10,0,0,1,-235.61046527773934,44.05354491698838L-107.22516425347698,21.34877792663628A10,10,0,0,1,-95.89427330027479,28.359977930389995A100,100,0,0,0,-55.66290755102201,83.07611403385678A10,10,0,0,1,-52.349182688810245,95.98212979487127Z"
              fill={
                Object {
                  "payload": 4282441936,
                  "type": 0,
                }
              }
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  375,
                  150,
                ]
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                Array [
                  "fill",
                ]
              }
              responsible={true}
            />
            <RNSVGPath
              d="M-236.4288457646136,39.425969342027614A10,10,0,0,1,-248.09407227894616,30.811220359619693A250,250,0,0,1,-104.0196343200738,-227.33217034972884A10,10,0,0,1,-90.56350710796434,-221.9262430368514L-42.488636540309855,-100.7359019073706A10,10,0,0,1,-47.07634398146938,-88.22594764205354A100,100,0,0,0,-99.80455305775081,6.2490950498934374A10,10,0,0,1,-108.04354474035118,16.72120235167547Z"
              fill={
                Object {
                  "payload": 4278255360,
                  "type": 0,
                }
              }
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  375,
                  150,
                ]
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                Array [
                  "fill",
                ]
              }
              responsible={true}
            />
            <RNSVGPath
              d="M-86.19526979374123,-223.6590745606583A10,10,0,0,1,-80.10409161920289,-236.81920214767712A250,250,0,0,1,-12.86426207573201,-249.66880213844678A10,10,0,0,1,-2.349691592702687,-239.68205005290892L-2.349691592702646,-109.30455213560472A10,10,0,0,1,-11.226992357002405,-99.36777466873157A100,100,0,0,0,-26.204597624345777,-96.50553902935381A10,10,0,0,1,-38.12039922608679,-102.46873343117747Z"
              fill={
                Object {
                  "payload": 4294613276,
                  "type": 0,
                }
              }
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  375,
                  150,
                ]
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                Array [
                  "fill",
                ]
              }
              responsible={true}
            />
          </RNSVGGroup>
          <RNSVGText
            content={null}
            dx={
              Array [
                "0",
              ]
            }
            dy={Array []}
            font={Object {}}
            rotate={Array []}
            x={
              Array [
                "375",
              ]
            }
            y={
              Array [
                "111.035",
              ]
            }
          >
            <RNSVGTSpan
              content="Net Worth"
              dx={
                Array [
                  "0",
                ]
              }
              dy={
                Array [
                  "0",
                ]
              }
              fill={
                Object {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                Object {
                  "fontFamily": "Gill Sans",
                  "fontSize": "17",
                  "textAnchor": "middle",
                }
              }
              propList={
                Array [
                  "fill",
                ]
              }
              rotate={Array []}
              x={
                Array [
                  "375",
                ]
              }
              y={Array []}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={
              Array [
                "0",
              ]
            }
            dy={Array []}
            font={Object {}}
            rotate={Array []}
            x={
              Array [
                "375",
              ]
            }
            y={
              Array [
                "134.585",
              ]
            }
          >
            <RNSVGTSpan
              content="Â£345"
              dx={
                Array [
                  "0",
                ]
              }
              dy={
                Array [
                  "0",
                ]
              }
              fill={
                Object {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                Object {
                  "fontFamily": "Gill Sans",
                  "fontSize": "27",
                  "fontWeight": "700",
                  "textAnchor": "middle",
                }
              }
              propList={
                Array [
                  "fill",
                ]
              }
              rotate={Array []}
              x={
                Array [
                  "375",
                ]
              }
              y={Array []}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={
              Array [
                "0",
              ]
            }
            dy={Array []}
            font={Object {}}
            rotate={Array []}
            x={
              Array [
                "375",
              ]
            }
            y={
              Array [
                "171.035",
              ]
            }
          >
            <RNSVGTSpan
              content="Assets"
              dx={
                Array [
                  "0",
                ]
              }
              dy={
                Array [
                  "0",
                ]
              }
              fill={
                Object {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                Object {
                  "fontFamily": "Gill Sans",
                  "fontSize": "17",
                  "textAnchor": "middle",
                }
              }
              propList={
                Array [
                  "fill",
                ]
              }
              rotate={Array []}
              x={
                Array [
                  "375",
                ]
              }
              y={Array []}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={
              Array [
                "0",
              ]
            }
            dy={Array []}
            font={Object {}}
            rotate={Array []}
            x={
              Array [
                "375",
              ]
            }
            y={
              Array [
                "194.585",
              ]
            }
          >
            <RNSVGTSpan
              content="4"
              dx={
                Array [
                  "0",
                ]
              }
              dy={
                Array [
                  "0",
                ]
              }
              fill={
                Object {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              font={
                Object {
                  "fontFamily": "Gill Sans",
                  "fontSize": "27",
                  "fontWeight": "700",
                  "textAnchor": "middle",
                }
              }
              propList={
                Array [
                  "fill",
                ]
              }
              rotate={Array []}
              x={
                Array [
                  "375",
                ]
              }
              y={Array []}
            />
          </RNSVGText>
          <View
            pointerEvents="box-none"
            style={
              Object {
                "left": 0,
                "position": "absolute",
                "top": 0,
                "zIndex": 99,
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="300"
              bbWidth="750"
              focusable={false}
              height={300}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "height": "100%",
                    "overflow": "visible",
                    "width": "100%",
                  },
                  Object {
                    "flex": 0,
                    "height": 300,
                    "width": 750,
                  },
                ]
              }
              vbHeight={300}
              vbWidth={750}
              width={750}
            >
              <RNSVGGroup />
            </RNSVGSvgView>
          </View>
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      className="VictoryContainer"
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="240"
        bbWidth="750"
        focusable={false}
        height={240}
        meetOrSlice={0}
        minX={0}
        minY={0}
        role="img"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
            Object {
              "flex": 0,
              "height": 240,
              "width": 750,
            },
          ]
        }
        vbHeight={240}
        vbWidth={750}
        width={750}
      >
        <RNSVGGroup>
          <RNSVGRect
            fill={null}
            height="240"
            propList={
              Array [
                "fill",
              ]
            }
            width="750"
            x="0"
            y="0"
          />
          <RNSVGGroup>
            <RNSVGPath
              d="M 40, 209 
A 0 0 0 0 1, 40, 209 
L 40, 191 
A 0 0 0 0 1, 40, 191 
L 710, 191 
A 0 0 0 0 1, 710, 191 
L 710, 209 
A 0 0 0 0 1, 710, 209 
 z"
              fill={
                Object {
                  "payload": 4294951115,
                  "type": 0,
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              responsible={true}
              stroke={
                Object {
                  "payload": 4294951115,
                  "type": 0,
                }
              }
              strokeWidth="0"
            />
            <RNSVGPath
              d="M 40, 155.66666666666669 
A 0 0 0 0 1, 40, 155.66666666666669 
L 40, 137.66666666666669 
A 0 0 0 0 1, 40, 137.66666666666669 
L 207.5, 137.66666666666669 
A 0 0 0 0 1, 207.5, 137.66666666666669 
L 207.5, 155.66666666666669 
A 0 0 0 0 1, 207.5, 155.66666666666669 
 z"
              fill={
                Object {
                  "payload": 4282441936,
                  "type": 0,
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              responsible={true}
              stroke={
                Object {
                  "payload": 4282441936,
                  "type": 0,
                }
              }
              strokeWidth="0"
            />
            <RNSVGPath
              d="M 40, 102.33333333333334 
A 0 0 0 0 1, 40, 102.33333333333334 
L 40, 84.33333333333334 
A 0 0 0 0 1, 40, 84.33333333333334 
L 291.25, 84.33333333333334 
A 0 0 0 0 1, 291.25, 84.33333333333334 
L 291.25, 102.33333333333334 
A 0 0 0 0 1, 291.25, 102.33333333333334 
 z"
              fill={
                Object {
                  "payload": 4278255360,
                  "type": 0,
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              responsible={true}
              stroke={
                Object {
                  "payload": 4278255360,
                  "type": 0,
                }
              }
              strokeWidth="0"
            />
            <RNSVGPath
              d="M 40, 49 
A 0 0 0 0 1, 40, 49 
L 40, 31 
A 0 0 0 0 1, 40, 31 
L 107, 31 
A 0 0 0 0 1, 107, 31 
L 107, 49 
A 0 0 0 0 1, 107, 49 
 z"
              fill={
                Object {
                  "payload": 4294613276,
                  "type": 0,
                }
              }
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              responsible={true}
              stroke={
                Object {
                  "payload": 4294613276,
                  "type": 0,
                }
              }
              strokeWidth="0"
            />
            <RNSVGText
              content={null}
              dx={
                Array [
                  "10",
                ]
              }
              dy={Array []}
              font={Object {}}
              name="bar-labels-0"
              rotate={Array []}
              x={
                Array [
                  "30",
                ]
              }
              y={
                Array [
                  "187.81",
                ]
              }
            >
              <RNSVGTSpan
                content="Credit Card"
                dx={
                  Array [
                    "10",
                  ]
                }
                dy={
                  Array [
                    "0",
                  ]
                }
                fill={
                  Object {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                font={
                  Object {
                    "fontFamily": "Gill Sans",
                    "fontSize": "22",
                    "fontWeight": "900",
                    "textAnchor": "start",
                  }
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotate={Array []}
                x={
                  Array [
                    "30",
                  ]
                }
                y={Array []}
              />
            </RNSVGText>
            <RNSVGText
              content={null}
              dx={
                Array [
                  "10",
                ]
              }
              dy={Array []}
              font={Object {}}
              name="bar-labels-1"
              rotate={Array []}
              x={
                Array [
                  "30",
                ]
              }
              y={
                Array [
                  "134.4766666666667",
                ]
              }
            >
              <RNSVGTSpan
                content="Crypto"
                dx={
                  Array [
                    "10",
                  ]
                }
                dy={
                  Array [
                    "0",
                  ]
                }
                fill={
                  Object {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                font={
                  Object {
                    "fontFamily": "Gill Sans",
                    "fontSize": "22",
                    "fontWeight": "900",
                    "textAnchor": "start",
                  }
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotate={Array []}
                x={
                  Array [
                    "30",
                  ]
                }
                y={Array []}
              />
            </RNSVGText>
            <RNSVGText
              content={null}
              dx={
                Array [
                  "10",
                ]
              }
              dy={Array []}
              font={Object {}}
              name="bar-labels-2"
              rotate={Array []}
              x={
                Array [
                  "30",
                ]
              }
              y={
                Array [
                  "81.14333333333335",
                ]
              }
            >
              <RNSVGTSpan
                content="Stocks"
                dx={
                  Array [
                    "10",
                  ]
                }
                dy={
                  Array [
                    "0",
                  ]
                }
                fill={
                  Object {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                font={
                  Object {
                    "fontFamily": "Gill Sans",
                    "fontSize": "22",
                    "fontWeight": "900",
                    "textAnchor": "start",
                  }
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotate={Array []}
                x={
                  Array [
                    "30",
                  ]
                }
                y={Array []}
              />
            </RNSVGText>
            <RNSVGText
              content={null}
              dx={
                Array [
                  "10",
                ]
              }
              dy={Array []}
              font={Object {}}
              name="bar-labels-3"
              rotate={Array []}
              x={
                Array [
                  "30",
                ]
              }
              y={
                Array [
                  "27.81",
                ]
              }
            >
              <RNSVGTSpan
                content="Debit Card"
                dx={
                  Array [
                    "10",
                  ]
                }
                dy={
                  Array [
                    "0",
                  ]
                }
                fill={
                  Object {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                font={
                  Object {
                    "fontFamily": "Gill Sans",
                    "fontSize": "22",
                    "fontWeight": "900",
                    "textAnchor": "start",
                  }
                }
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotate={Array []}
                x={
                  Array [
                    "30",
                  ]
                }
                y={Array []}
              />
            </RNSVGText>
          </RNSVGGroup>
          <View
            pointerEvents="box-none"
            style={
              Object {
                "left": 0,
                "position": "absolute",
                "top": 0,
                "zIndex": 99,
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="240"
              bbWidth="750"
              focusable={false}
              height={240}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "height": "100%",
                    "overflow": "visible",
                    "width": "100%",
                  },
                  Object {
                    "flex": 0,
                    "height": 240,
                    "width": 750,
                  },
                ]
              }
              vbHeight={240}
              vbWidth={750}
              width={750}
            >
              <RNSVGGroup />
            </RNSVGSvgView>
          </View>
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "borderRadius": 25,
              "fontSize": 30,
              "fontWeight": "bold",
              "textAlign": "center",
              "width": "75%",
            },
            Object {
              "color": "black",
            },
          ]
        }
        x={10}
      >
        Back
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
